apps/opik-frontend/src/components/redirect/RedirectProjects.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useMemo } from "react";
+import Loader from "@/components/shared/Loader/Loader";
+import { StringParam, useQueryParams } from "use-query-params";
+import useAppStore from "@/store/AppStore";
+import { useNavigate } from "@tanstack/react-router";
+import useProjectByName from "@/api/projects/useProjectByName";
+
+const RedirectProjects = () => {
```

Comment:
I think we can simplify a bit of logic in this component. 
1) In case we have `id` we redirect immediately. 
2) If we don't have it but have `name`, we try to get it by calling API and redirecting after that.